src/domain/datasets/services/src/dataset_key_value_service_sys_env.rs

```diff
@@ -35,15 +36,28 @@ impl DatasetKeyValueServiceSysEnv {
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-#[async_trait::async_trait]
 impl DatasetKeyValueService for DatasetKeyValueServiceSysEnv {
-    async fn find_dataset_env_var_value_by_key<'a>(
+    fn find_dataset_env_var_value_by_key(
         &self,
         dataset_env_var_key: &str,
-        _dataset_env_vars: &'a HashMap<String, DatasetEnvVar>,
+        dataset_env_vars: &HashMap<String, DatasetEnvVar>,
     ) -> Result<DatasetEnvVarValue, FindDatasetEnvVarError> {
+        // Search explicitly provided vars first
+        if let Some(var) = dataset_env_vars.get(dataset_env_var_key) {
+            return Ok(DatasetEnvVarValue::Regular(
+                var.get_non_secret_value()
+                    .expect("Unexpected encrypted value"),
+            ));
+        }
+
+        // Fall back to system environment
```

Comment:
Not sure it is safe to put fallback logic where we fetch env from env vars.
As we discussed we shouldn't allow API to work with secrets as env vars. Let's say API will require some envs like `PORT` etc and in the dataset, we will have a similar secret. If the user forgets to set such a value it will be read from env vars and it will be hard to identify

Reply:
This change is not adding a fallback to read from sys env, but rather adding an option to read values from expliciltly-defined variables in the HashMap before fetching from sys env.

The `DatasetKeyValueServiceImpl` is the one that should be used in `api-server` and that one should never fall back to sys env.